api/models/series.go

```diff
@@ -26,19 +26,46 @@ type Series struct {
 }
 
 func (s *Series) SetTags() {
-	tagSplits := strings.Split(s.Target, ";")
+	numTags := strings.Count(s.Target, ";")
+
+	if s.Tags == nil {
+		// +1 for the name tag
+		s.Tags = make(map[string]string, numTags)
```

Comment:
don't need to use numTags+1 per the comment?

Reply:
Good catch. Must have gotten lost during some of my benchmark rounds